src/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str.cs

```diff
@@ -1,221 +1,219 @@
+        [Fact]
+        public void NonexistentPath()
         {
-            String[] results = GetEntries(TestDirectory);
-            Assert.Contains(testFile1, results);
-            Assert.DoesNotContain(testFile2, results);
+            Assert.Throws<DirectoryNotFoundException>(() => GetEntries(GetTestFilePath()));
         }
-    }
 
-    #endregion
+        [Fact]
+        public void TrailingSlashes()
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            Directory.CreateDirectory(Path.Combine(testDir.FullName, GetTestFileName()));
+            using (File.Create(Path.Combine(testDir.FullName, GetTestFileName())))
+            {
+                string[] strArr = GetEntries(testDir.FullName + new string(Path.DirectorySeparatorChar, 5));
+                Assert.NotNull(strArr);
+                Assert.NotEmpty(strArr);
+            }
+        }
 
-    #region PlatformSpecific
+        [Fact]
+        public void CurrentDirectory()
+        {
+            Assert.NotNull(GetEntries(Directory.GetCurrentDirectory()));
+        }
 
-    [Fact]
-    [PlatformSpecific(PlatformID.Windows)]
-    public void WildCharactersFileName()
-    {
-        String strTempDir = Path.Combine("dls;d", "442349-0", "v443094(*)(+*$#$*") + new string(Path.DirectorySeparatorChar, 3);
-        Assert.Throws<ArgumentException>(() => GetEntries(strTempDir));
-    }
+        #endregion
 
-    [Fact]
-    [PlatformSpecific(PlatformID.Windows)]
-    public void WindowsFileNameWithSpaces()
-    {
-        Assert.Throws<ArgumentException>(() => GetEntries("\n"));
-        Assert.Throws<ArgumentException>(() => GetEntries("          "));
-        Assert.Throws<ArgumentException>(() => GetEntries(" "));
-        Assert.Throws<ArgumentException>(() => GetEntries(""));
-        Assert.Throws<ArgumentException>(() => GetEntries(">"));
-        Assert.Throws<ArgumentException>(() => GetEntries("\0"));
-    }
+        #region PlatformSpecific
 
-    [Fact]
-    [PlatformSpecific(PlatformID.AnyUnix)]
-    public void UnixFileNameWithSpaces()
-    {
-        String testDirPath = Path.Combine(TestDirectory, GetTestFileName());
-        Directory.CreateDirectory(testDirPath);
-        using (File.Create(Path.Combine(testDirPath, " ")))
-        using (File.Create(Path.Combine(testDirPath, "          ")))
-        using (File.Create(Path.Combine(testDirPath, "\n")))
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void WindowsWildCharactersFileName()
         {
-            String[] results = GetEntries(testDirPath);
-            Assert.Contains(Path.Combine(testDirPath, " "), results);
-            Assert.Contains(Path.Combine(testDirPath, "          "), results);
-            Assert.Contains(Path.Combine(testDirPath, "\n"), results);
+            string strTempDir = Path.Combine("dls;d", "442349-0", "v443094(*)(+*$#$*") + new string(Path.DirectorySeparatorChar, 3);
```

Comment:
I actually think I'll remove this test and replace it with something that calls Path.GetInvalidFileNameChars so it can be more xplat friendly. I copied this one from the old code before adding new tests that mostly eclipsed its original purpose.